modules/sdk/gradle-plugins-osgi-bundle-builder/src/main/java/com/liferay/gradle/plugins/osgi/bundle/builder/OSGiBundleBuilderPlugin.java

```diff
@@ -0,0 +1,187 @@
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.gradle.plugins.osgi.bundle.builder;
+
+import com.liferay.gradle.util.GradleUtil;
+
+import java.io.File;
+
+import java.util.concurrent.Callable;
+
+import org.gradle.api.Action;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.artifacts.DependencySet;
+import org.gradle.api.file.FileCollection;
+import org.gradle.api.plugins.BasePlugin;
+import org.gradle.api.plugins.JavaPlugin;
+import org.gradle.api.tasks.Copy;
+import org.gradle.api.tasks.SourceSet;
+import org.gradle.api.tasks.SourceSetOutput;
+import org.gradle.api.tasks.TaskContainer;
+import org.gradle.api.tasks.compile.JavaCompile;
+
+/**
+ * @author David Truong
+ */
+public class OSGiBundleBuilderPlugin implements Plugin<Project> {
+
+	public static final String EXPLODED_JAR_TASK_NAME = "explodedJar";
+
+	public static final String OSGI_BUNDLE_BUILDER_CONFIGURATION_NAME =
+		"osgiBundleBuilder";
+
+	@Override
+	public void apply(Project project) {
+		Configuration osgiBundleBuilderConfiguration =
+			_addConfigurationOSGiBundleBuilder(project);
+
+		_addTaskExplodedJar(project);
+
+		_configureTasksExplodedJar(project, osgiBundleBuilderConfiguration);
+	}
+
+	private Configuration _addConfigurationOSGiBundleBuilder(
+		final Project project) {
+
+		Configuration configuration = GradleUtil.addConfiguration(
+			project, OSGI_BUNDLE_BUILDER_CONFIGURATION_NAME);
+
+		configuration.defaultDependencies(
+			new Action<DependencySet>() {
+
+				@Override
+				public void execute(DependencySet dependencySet) {
+					_addDependenciesOSGiBundleBuilder(project);
+				}
+
+			});
+
+		configuration.setDescription(
+			"Configures Liferay OSGi Bundle Builder for this project.");
+		configuration.setVisible(false);
+
+		return configuration;
+	}
+
+	private void _addDependenciesOSGiBundleBuilder(Project project) {
+		GradleUtil.addDependency(
+			project, OSGI_BUNDLE_BUILDER_CONFIGURATION_NAME, "com.liferay",
+			"com.liferay.osgi.bundle.builder", "latest.release");
+	}
+
+	private ExplodedJarTask _addTaskExplodedJar(Project project) {
+		final SourceSet sourceSet = GradleUtil.getSourceSet(
+			project, SourceSet.MAIN_SOURCE_SET_NAME);
+
+		final SourceSetOutput sourceSetOutput = sourceSet.getOutput();
+
+		final ExplodedJarTask explodedJarTask = GradleUtil.addTask(
+			project, EXPLODED_JAR_TASK_NAME, ExplodedJarTask.class);
+
+		JavaCompile javaCompile = (JavaCompile)GradleUtil.getTask(
+			project, JavaPlugin.COMPILE_JAVA_TASK_NAME);
+
+		Copy processResources = (Copy)GradleUtil.getTask(
+			project, JavaPlugin.PROCESS_RESOURCES_TASK_NAME);
+
+		explodedJarTask.dependsOn(javaCompile, processResources);
```

Comment:
Actually, why don't you just depend on the `classes` task? Use [this](https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/SourceSet.html#getClassesTaskName--) to get the task name.